src/java/org/apache/cassandra/transport/Message.java

```diff
@@ -215,10 +222,33 @@ protected boolean isTraceable()
             return false;
         }
 
+        protected long elapsedTimeSinceCreation(TimeUnit timeUnit)
+        {
+            return timeUnit.convert(MonotonicClock.approxTime.now() - creationTimeNanos, TimeUnit.NANOSECONDS);
+        }
+
+        protected void markExecutingAsync()
+        {
+            isExecutingAsync = true;
+        }
+
+        protected boolean isExecutingAsync()
+        {
+            return isExecutingAsync;
+        }
+
         protected abstract CompletableFuture<Response> maybeExecuteAsync(QueryState queryState, long queryStartNanoTime, boolean traceRequest);
 
         public final CompletableFuture<Response> execute(QueryState queryState, long queryStartNanoTime)
         {
+            long elapsedTimeSinceCreation = elapsedTimeSinceCreation(TimeUnit.NANOSECONDS);
```

Comment:
So async queue time includes sync queue time? Could you please make the relation clear in the comments?